bundles/org.eclipse.epsilon.emc.rdf/src/org/eclipse/epsilon/emc/rdf/RDFModel.java

```diff
@@ -368,15 +372,42 @@ protected void validateModel() throws Exception {
 		 */
 
 		ValidityReport modelValidityReport = model.validate();
-		if (!modelValidityReport.isValid() || !modelValidityReport.isClean()) {
-			StringBuilder sb = new StringBuilder("The loaded model is not valid or not clean\n");
-			int i = 1;
-			for (Iterator<Report> o = modelValidityReport.getReports(); o.hasNext();) {
-				ValidityReport.Report report = (ValidityReport.Report) o.next();
-				sb.append(String.format(" %d: %s", i, report.toString()));
-				i++;
+
+		switch (validationMode) {
+		case VALIDATION_SELECTION_JENA_CLEAN:
+		case VALIDATION_SELECTION_JENA_VALID:
+			if (!modelValidityReport.isValid() || !modelValidityReport.isClean()) {
+				StringBuilder sb = new StringBuilder("The loaded model is ");
+
+				if (!modelValidityReport.isValid()) {
+					sb.append("not ");
+				}
+				sb.append("valid");
+
+				if (validationMode.equals(VALIDATION_SELECTION_JENA_CLEAN)) {
+					sb.append(" and ");
+					if (!modelValidityReport.isClean()) {
+						sb.append("not ");
+					}
+					sb.append("clean");
+				}
+				
+				sb.append("\n");
+				
+				// Build report string (valid models still report warnings)
+				int i = 1;
+				for (Iterator<Report> o = modelValidityReport.getReports(); o.hasNext();) {
+					ValidityReport.Report report = (ValidityReport.Report) o.next();
+					sb.append(String.format(" %d: %s", i, report.toString()));
+					i++;
+				}
+				
+				throw new Exception(sb.toString());
```

Comment:
Should be a new custom `RDFValidationException` type - let's make it a runtime exception (so subclass of `RuntimeException`).